lib/console_web/controllers/gateway_controller.ex

```diff
@@ -18,7 +18,9 @@ defmodule ConsoleWeb.GatewayController do
   def create(conn, %{"gateway" => gateway_params}) do
     current_user = conn.assigns.current_user
     current_team = conn.assigns.current_team
-    gateway_params = Map.merge(gateway_params, %{"team_id" => current_team.id})
+    location = geocodeLatLng(gateway_params)
+
+    gateway_params = Map.merge(gateway_params, %{"team_id" => current_team.id, "location" => location})
```

Comment:
this should be in a changeset so that when lat/lng are changed, the location is changed as well

Reply:
good point